mne/decoding/csp.py

```diff
@@ -62,97 +66,131 @@ class CSP(TransformerMixin, EstimatorMixin):
         Klaus-Robert Müller. Optimizing Spatial Filters for Robust EEG
         Single-Trial Analysis. IEEE Signal Processing Magazine 25(1), 41-56,
         2008.
+    [3] Grosse-Wentrup, Moritz, and Martin Buss. Multiclass common spatial
+        patterns and information theoretic feature extraction. IEEE
+        Transactions on Biomedical Engineering, Vol 55, no. 8, 2008.
     """
 
     def __init__(self, n_components=4, reg=None, log=True, cov_est="concat"):
         """Init of CSP."""
+        if not isinstance(n_components, int):
+            raise ValueError('n_components must be an integer.')
         self.n_components = n_components
+        if (
+            (reg is not None) and
+            (reg not in ['oas', 'ledoit_wolf']) and
+            ((not isinstance(reg, (float, int))) or
+             (not ((reg <= 1.) and (reg >= 0.))))
+        ):
+            raise ValueError('reg must be None, "oas", "ledoit_wolf" or a '
+                             'float in between 0. and 1.')
         self.reg = reg
         self.log = log
+        if not (cov_est == "concat" or cov_est == "epoch"):
+            raise ValueError("unknown covariance estimation method")
         self.cov_est = cov_est
-        self.filters_ = None
-        self.patterns_ = None
-        self.mean_ = None
-        self.std_ = None
-
-    def get_params(self, deep=True):
-        """Return all parameters (mimics sklearn API).
 
-        Parameters
-        ----------
-        deep: boolean, optional
-            If True, will return the parameters for this estimator and
-            contained subobjects that are estimators.
-        """
-        params = {"n_components": self.n_components,
-                  "reg": self.reg,
-                  "log": self.log}
-        return params
+    def _check_Xy(self, X, y=None):
+        """Aux. function to check input data."""
+        if y is not None:
+            if len(X) != len(y) or len(y) < 1:
+                raise ValueError('X and y must have the same length.')
+        if X.ndim < 3:
+            raise ValueError('X must have at least 3 dimensions.')
 
-    def fit(self, epochs_data, y):
+    def fit(self, X, y, epochs_data=None):
         """Estimate the CSP decomposition on epochs.
 
         Parameters
         ----------
-        epochs_data : ndarray, shape (n_epochs, n_channels, n_times)
-            The data to estimate the CSP on.
+        X : ndarray, shape (n_epochs, n_channels, n_times)
+            The data on which to estimate the CSP.
         y : array, shape (n_epochs,)
             The class for each epoch.
+        epochs_data : None | ndarray, shape (n_epochs, n_channels, n_times)
+            The data on which to estimate the CSP. Overwrite X if not None.
+            Will be deprecated in mne 0.14.
```

Comment:
remove epochs_data it's deprecated. Don't keep it just for travis. Their must be a way to have it ignored